api/src/main/java/io/grpc/LoadBalancer.java

```diff
@@ -148,15 +150,17 @@ public void handleResolvedAddressGroups(
    * <p>Implementations should not modify the given {@code servers}.
    *
    * @param resolvedAddresses the resolved server addresses, attributes, and config.
+   * @return {@code true} if the resolved addresses were accepted. {@code false} if rejected.
    * @since 1.21.0
    */
   @SuppressWarnings("deprecation")
-  public void handleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
+  public boolean handleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
```

Comment:
With this change, we'd expect every load balancer to break (since it appears the other methods have been deprecated for a long time). I think we need to introduce a new method instead. If the user doesn't override the new method, let's have that method do the zero address check based on `canHandleEmptyAddressListFromNameResolution()` and emulate the old behavior (that's previously in AutoConfiguredLoadBalancer) by providing a failing picker and returning false.

Reply:
That's right, I can't just go changing it like that. I removed the old deprecated method, deprecated the current one and created a new one that returns a boolean.

Right now the default implementation of the new one just delegates to the existing one and returns true. This might not be correct, but I'm not sure what you mean by "based on `canHandleEmptyAddressListFromNameResolution()`", that method is getting removed in this PR. Maybe we should have a chat on this whole PR overall.

Reply:
We discussed this offline - the PR's focus has changed to deprecating `handleResolvedAddresses()` and will still keep `canHandleEmptyAddressListFromNameResolution()` around.